docs/source/backend.rst

```diff
@@ -3,44 +3,88 @@
+    * ``"sox"`` (deprecated)
+    * ``"sox_io"``
+    * ``"soundfile"``, legacy interface (deprecated)
+    * ``"soundfile"``, new interface
+
+``"sox"`` backend is the original backend which is built on ``libsox``. This backend requires C++ extension module and is not available on Windows system. This backend is currently the default backend when available, but is being deprecated. Please use ``"sox_io"`` backend.
 
-There are currently three implementations available.
+``"sox_io"`` backend is the new backend which is built on ``libsox``, and bound with ``TorchScript``. This backend requires C++ extension module and is not available on Windows system. Function calls to this backend are TorchScript-able. This backend will become the default backend in ``"0.8.0"`` release.
 
-    * :ref:`sox<sox_backend>`
-    * :ref:`sox_io<sox_io_backend>`
-    * :ref:`soundfile<soundfile_backend>`
+``"soundfile"`` backend is built on ``PySoundFile``. You need to install ``PySoundFile`` separately. The current interface of this backend (legacy interface) will be changed in ``0.8.0`` (new interface) to match the ``"sox_io"`` backend.
 
-``sox`` backend is the original backend which is built on ``libsox``. This module is currently default but is known to have number of issues, such as wrong handling of WAV files other than 16-bit signed integer. Users are encouraged to use ``sox_io`` backend. This backend requires C++ extension module and is not available on Windows system.
+.. note::
+   Instead of calling functions in :mod:`torchaudio.backend` directly, please use ``torchaudio.info``, ``torhcaudio.load``, ``torchaudio.load_wav`` and ``torchaudio.save`` with proper backend set with :func:`torchaudio.get_audio_backend`.
 
-``sox_io`` backend is the new backend which is built on ``libsox`` and bound to Python with ``Torchscript``. This module addresses all the known issues ``sox`` backend has. Function calls to this backend can be Torchscriptable. This backend requires C++ extension module and is not available on Windows system.
+Changes in default backend and Deprecation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-``soundfile`` backend is built on ``PySoundFile``. You need to install ``PySoundFile`` separately.
+.. warning::
+   * The default backend for Linux/macOS will be changed from ``"sox"`` to ``"sox_io"`` in ``0.8.0`` release.
+   * ``"sox"``  backend will be removed in ``0.9.0`` release.
+   * The function signatures of ``"soundfile"`` backends are changed in ``0.8.0`` to match ``"sox_io"`` backend.
+   * To opt-in to the new signature of ``"soundfile"`` backend, do ``torchaudio.USE_SOUNDFILE_LEGACY_INTERFACE = False`` **before** switching to ``"soundfile"``.
+
+   The following table summarizes the timeline for the changes and deprecations.
```

Comment:
This is great! Want to make this part of an issue as well?

Reply:
Do we want to write a migration guide tutorial?

Reply:
> This is great! Want to make this part of an issue as well?

[previous response](https://github.com/pytorch/audio/pull/935#discussion_r499009228)